agent/lib_aws_sdk_php.c

```diff
@@ -14,10 +14,405 @@
+            'StringListValues' => array('string', ... ),
+            'BinaryListValues' => array('string', ... ),
+            // DataType is required
+            'DataType' => 'string',
+        ),
+        // ... repeated
+    ),
+));
+
+$result = $client->sendMessageBatch(array(
+    // QueueUrl is required
+    'QueueUrl' => 'string',
+    // Entries is required
+    'Entries' => array(
+        array(
+            // Id is required
+            'Id' => 'string',
+            // MessageBody is required
+            'MessageBody' => 'string',
+            'DelaySeconds' => integer,
+            'MessageAttributes' => array(
+                // Associative array of custom 'String' key names
+                'String' => array(
+                    'StringValue' => 'string',
+                    'BinaryValue' => 'string',
+                    'StringListValues' => array('string', ... ),
+                    'BinaryListValues' => array('string', ... ),
+                    // DataType is required
+                    'DataType' => 'string',
+                ),
+                // ... repeated
+            ),
+        ),
+        // ... repeated
+    ),
+));
+
+//clang-format on
+*/
+void nr_lib_aws_sdk_php_sqs_handle(nr_segment_t* segment,
+                                   char* command_name_string,
+                                   size_t command_name_len,
+                                   NR_EXECUTE_PROTO) {
+  char* command_arg_value = NULL;
+  nr_segment_message_params_t message_params = {
+      .library = "SQS",
+      .destination_type = NR_MESSAGE_DESTINATION_TYPE_QUEUE,
+      .messaging_system = "aws_sqs",
+  };
+  nr_segment_cloud_attrs_t cloud_attrs = {0};
+
+  if (NULL == segment) {
+    return;
+  }
+
+  if (NULL == command_name_string || 0 == command_name_len) {
+    return;
+  }
+
+#define AWS_COMMAND_IS(CMD) \
+  (command_name_len == (sizeof(CMD) - 1) && nr_streq(CMD, command_name_string))
+
+  /* Determine if we instrument this command. */
+  if (AWS_COMMAND_IS("sendMessageBatch")) {
+    message_params.message_action = NR_SPANKIND_PRODUCER;
+  } else if (AWS_COMMAND_IS("sendMessage")) {
+    message_params.message_action = NR_SPANKIND_PRODUCER;
+  } else if (AWS_COMMAND_IS("receiveMessage")) {
+    message_params.message_action = NR_SPANKIND_CONSUMER;
+  } else {
+    /* Nothing to do here so exit. */
+    return;
+  }
+#undef IS_AWS_COMMAND
+
+  cloud_attrs.aws_operation = command_name_string;
+
+  command_arg_value = nr_lib_aws_sdk_php_get_command_arg_value(
+      AWS_SDK_PHP_SQSCLIENT_QUEUEURL_ARG, NR_EXECUTE_ORIG_ARGS);
+
+  /*
+   * nr_lib_aws_sdk_php_sqs_parse_queueurl requires a modifiable string to
+   * populate message_params and cloud_attrs.
+   */
+  nr_lib_aws_sdk_php_sqs_parse_queueurl(command_arg_value, &message_params,
+                                        &cloud_attrs);
+
+  /* Add cloud attributes, if available. */
+
+  nr_segment_traces_add_cloud_attributes(segment, &cloud_attrs);
```

Comment:
Just for posterity - no `char *` pointers in `could_attrs` need to be freed because they're either pointers into `command_arg_value` which is freed before returning from this function or it is a pointer to `command_name_string` which is also freed.